plasma_framework/test/src/exits/payment/PaymentChallengeIFEOutput.test.js

```diff
@@ -0,0 +1,295 @@
+                    exitTarget: alice,
+                    token: ETH,
+                    amount: AMOUNT,
+                }, DUMMY_WITHDRAW_DATA, DUMMY_WITHDRAW_DATA, DUMMY_WITHDRAW_DATA],
+                outputs: [{
+                    outputId: filler,
+                    outputGuard: filler,
+                    exitTarget: alice,
+                    token: ETH,
+                    amount: AMOUNT,
+                }, {
+                    outputId: filler,
+                    outputGuard: filler,
+                    exitTarget: alice,
+                    token: ETH,
+                    amount: AMOUNT,
+                }, DUMMY_WITHDRAW_DATA, DUMMY_WITHDRAW_DATA],
+            };
+
+            return {
+                exitId,
+                inFlightExit,
+                inFlightTxBytes,
+                challengingTxBytes,
+                blockRoot: merkleTree.root,
+                inclusionProof,
+                exitingOutputUtxoPos: buildUtxoPos(BLOCK_NUM, 0, 0),
+            };
+        };
+
+        beforeEach(async () => {
+            this.outputGuardHandlerRegistry = await OutputGuardHandlerRegistry.new();
+            const expectedOutputGuardHandler = await OutputGuardHandler.new();
+            await expectedOutputGuardHandler.mockIsValid(true);
+            await expectedOutputGuardHandler.mockGetExitTarget(alice);
+            await this.outputGuardHandlerRegistry.registerOutputGuardHandler(
+                OUTPUT_TYPE_ONE, expectedOutputGuardHandler.address,
+            );
+
+            this.framework = await SpyPlasmaFramework.new(
+                MIN_EXIT_PERIOD, DUMMY_INITIAL_IMMUNE_VAULTS_NUM, INITIAL_IMMUNE_EXIT_GAME_NUM,
+            );
+
+            this.spendingConditionRegistry = await PaymentSpendingConditionRegistry.new();
+            const conditionTrue = await PaymentSpendingConditionTrue.new();
+            await this.spendingConditionRegistry.registerSpendingCondition(
+                OUTPUT_TYPE_ONE, IFE_TX_TYPE, conditionTrue.address,
+            );
+
+            this.exitGame = await PaymentInFlightExitRouter.new(
+                this.framework.address,
+                this.outputGuardHandlerRegistry.address,
+                this.spendingConditionRegistry.address,
+                this.stateTransitionVerifierAccept.address,
+                IFE_TX_TYPE,
+            );
+
+            const args = await buildValidChallengeOutputArgs();
+
+            await this.framework.setBlock(BLOCK_NUM, args.blockRoot, 0);
+            await this.exitGame.setInFlightExit(args.exitId, args.inFlightExit);
+
+            const preimage = web3.utils.bytesToHex('preimage');
+            const piggybackArgs = {
+                inFlightTx: args.inFlightTxBytes,
+                outputIndex: 0,
+                outputType: OUTPUT_TYPE_ONE,
+                outputGuardPreimage: preimage,
+            };
+
+            await this.exitGame.piggybackInFlightExitOnOutput(
```

Comment:
If we would want to do proper unit tests, then we should probably code a mock that allows to "set" piggybacks.
I've decided to use a true piggyback because:
1. It saves time for coding and maintaining the mock.
2. Challenges are very tightly coupled with piggybacks.
If you have a different opinion on that then I'm open to changing that.

Reply:
await this.exitGame.setInFlightExit(args.exitId, args.inFlightExit);

You’re stubbing data already anyway, wouldn’t it be easier to stub the exit map for piggyback flag directly?

Reply:
:man_facepalming: you are right, I'll fix that with the other changes proposed on PR.